The previous section show write exception handler writeList method ListOfNumbers class . Sometimes , 's appropriate code catch exception occur within . In case , however , 's good let method far call stack handle exception . For example , provide ListOfNumbers class part package class , probably could n't anticipate need user package . In case , 's good catch exception allow method far call stack handle . If writeList method n't catch checked exception occur within , writeList method must specify throw exception . Let 's modify original writeList method specify exception throw instead catch . To remind , 's original version writeList method wo n't compile . public void writeList ( ) { PrintWriter = new PrintWriter ( new FileWriter ( `` OutFile.txt '' ) ) ; ( int = 0 ; < SIZE ; i++ ) { out.println ( `` Value : `` + + `` = `` + list.get ( ) ) ; } out.close ( ) ; } To specify writeList throw two exception , add throw clause method declaration writeList method . The throw clause comprise throw keyword follow comma-separated list exception throw method . The clause go method name argument list brace define scope method ; 's example . public void writeList ( ) throw IOException , IndexOutOfBoundsException { Remember IndexOutOfBoundsException unchecked exception ; include throw clause mandatory . You could write following . public void writeList ( ) throw IOException {